### PR TITLE
Full flow upgrade testing

### PR DESCRIPTION
**Pull request by nolancon**

### CODE REVIEW

File: e2e/tests/1.21/upgrade/07-write-data.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+  - command: ../../../../hack/write-pod-data.sh pod-1 /mnt/storageosupgrade StorageOS
```
Comment (Mojachieee) [inline e2e/tests/1.21/upgrade/07-write-data.yaml:4]: Is there any way to know if this failed?

As we don't have an assert for it

Comment (Mojachieee) [reply e2e/tests/1.21/upgrade/07-write-data.yaml:4]: Ah I see now that we later read the data and check it matches

**Review by Mojachieee — APPROVED** [review]

File: e2e/tests/1.21/upgrade/07-write-data.yaml (hunk 1 above)
Comment (nolancon) [reply e2e/tests/1.21/upgrade/07-write-data.yaml:4]: If a kuttl command fails to execute, the test-step will automatically fail and exit from what I've seen.

**Review by mhmxs — COMMENTED** [review]
LGTM

**Pull request merged**